pyronear_ds/preprocess/final_merger/same_time_span_selector.py

```diff
@@ -0,0 +1,46 @@
+from typing import Tuple
+
+import pandas as pd
+from pandas import DatetimeIndex, DataFrame
+
+from pyronear_ds.preprocess.geographic_data import GeographicData
+
+
+class SameTimeSpanSelector:
+    @staticmethod
+    def compute_time_span(dataframe: DataFrame, time_col: str):
+        min_time, max_time = dataframe[time_col].min(), dataframe[time_col].max()
+        return min_time, max_time
+
+    def find_largest_time_span(
+            self, dataclass1: GeographicData, dataclass2: GeographicData
+    ) -> DatetimeIndex:
+        min_time1, max_time1 = self.compute_time_span(
+            dataclass1.dataframe, dataclass1.time_col
+        )
+        time_span1 = pd.date_range(min_time1, max_time1)
+        min_time2, max_time2 = self.compute_time_span(
+            dataclass2.dataframe, dataclass2.time_col
+        )
+        time_span2 = pd.date_range(min_time2, max_time2)
+        if len(time_span1) > len(time_span2):
+            return time_span2
+        else:
+            return time_span1
```

Comment:
Not entirely sure, but if the goals is to find the largest time span, shouldn't this be replaced with :
`return time_span1 if time_span1 > time_span2 else time_span2` ?

This seems to be selecting the shortest time span :thinking:

Reply:
No, it should select the intersection of both time spans. Therefore we should find the smallest time span, so that both dataset cover the same time span (which has to be the smallest time span). Need the name is confusing! Going do change it